src/store.rs

```diff
@@ -0,0 +1,42 @@
+use std::path::{Path, PathBuf};
+use regex::Regex;
+
+const HASH_REG: &str = "^([[:alnum:]]+):([[:xdigit:]]+)$";
+
+pub fn hash_parse(hash: &str) -> (&str, &str) {
+    let hash_reg = Regex::new(HASH_REG).unwrap();
+    let caps = hash_reg.captures(hash).unwrap();
+    (caps.get(1).map_or("", |m| m.as_str()),
```

Comment:
no test below of the `""` behaviour here - worth testing? worth returning a Result type?

Reply:
Yeah maybe better to return `Result` actually, since otherwise you'll just get an error at the next step with a message that doesn't make loads of sense. Will change over and test :+1: